File: backend/Multibanca.Domain.Models/Multibanca/rectificatoria_firma.cs
using Common.Domain.Models;

namespace Multibanca.Domain.Models.Multibanca
{
    public class rectificatoria_firma : base_auditoria
    {
        public int id_rectificatoria_firma { get; set; }

        public long id_expediente { get; set; }

        public int id_usuario_solicitante { get; set; }

        public bool is_subsanar { get; set; }

        public string? observaciones { get; set; }

        // Campos enriquecidos para pantalla. No existen en la tabla física.
        public string? solicitante { get; set; }

        public string? observaciones_reparo { get; set; }

        public DateTime? fecha_ingreso { get; set; }

        public List<rectificatoria_firma_detalle>? detalles { get; set; }
    }
}

File: frontend/src/features/actividades/rectificatoria_firma/models/rectificatoria_firma.ts
import { RectificatoriaFirmaDetalle } from "./rectificatoria_firma_detalle";

export interface RectificatoriaFirma {
  id_rectificatoria_firma: number;
  id_expediente: number;
  id_usuario_solicitante: number;
  is_subsanar: boolean;
  observaciones?: string | null;

  /**
   * Campos heredados/visualización.
   * Vienen desde el reparo generado en Visar Operación.
   */
  solicitante?: string | null;
  observaciones_reparo?: string | null;
  fecha_ingreso?: string | null;

  is_active: boolean;
  row_status: boolean;
  created_by: number;
  created_date: string;
  modified_by?: number | null;
  modified_date?: string | null;

  detalles?: RectificatoriaFirmaDetalle[];
}
